templates/slim/block_table.html.slim

```diff
@@ -24,9 +25,10 @@ table(id=@id class=['tableblock',"frame-#{attr :frame, 'all'}","grid-#{attr :gri
                   - cell_content = cell.text
                 - else
                   - cell_content = cell.content
-              *{:tag=>(tblsec == :head ? 'th' : 'td'), :class=>['tableblock',"halign-#{cell.attr :halign}","valign-#{cell.attr :valign}"],
+              = html_tag(tblsec == :head || cell.style == :header ? 'th' : 'td',
```

Comment:
`html_tag` is not provided by Asciidoctor, it’s defined in my `helpers.rb`:

https://github.com/jirutka/asciidoctor-html5s/blob/a71db48a1dd5196b668b3a3d93693c5d877c5bf3/data/templates/helpers.rb#L84-L108

I was thinking about creating Slim/Temple extension like [slim-htag](https://github.com/jirutka/slim-htag), but for arbitrary tag. Actually I’ve implemented it and it works, but I don’t know how to name it (and how to name the special attribute) and if it’s really a good thing to do: it’s not idiomatic and it does break The Principle of Least Surprise.

It may look like this:

```rb
??? [
  ??=(tblsec == :head || cell.style == :header ? 'th' : 'td')
  class=['tableblock', "halign-#{cell.attr :halign}", "valign-#{cell.attr :valign}"]
  colspan=cell.colspan
  rowspan=cell.rowspan
  style=((@document.attr? :cellbgcolor) ? %(background-color:#{@document.attr :cellbgcolor};) : nil) ]
```

`???` should be name of the special “tag” (e.g. `element`, `dyntag`, …?) and `??` name of the special attribute defining the actual tag (e.g. `tag`, `_`, …?).

Reply:
After quickly looking at [HTML documentation](https://html.spec.whatwg.org/multipage/semantics.html#semantics) I would say `???` be `html_element` and `??` = `tag_name` but that's far from compact. `dyntag` and `tag` would work for me as shorter alternatives.

Reply:
In the meantime, I've added your `html_tag` function to our `helpers.rb`